File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.java
class Solution {
    public int[] leftRightDifference(int[] nums) {
        int n = nums.length;
        int[] pre = new int[n];
        int [] post = new int[n];
        for(int i=1;i<n;i++){
            pre[i]=pre[i-1]+nums[i-1];
        }
        for(int i=n-2;i>=0;i--){
            post[i]=post[i+1]+nums[i+1];
        }

        int [] ans = new int[n];
        for(int i=0;i<n;i++){
            ans[i]= Math.abs(pre[i]-post[i]);
        }
        // System.out.println(Arrays.toString(pre));
        //  System.out.println(Arrays.toString(post));
        return ans;
    }
}

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        n = len(nums)
        pre=[0]*n
        post = [0]*n
        ans = [0]*n
        
        idx = 1
        while idx<n:
            pre[idx] = pre[idx-1]+nums[idx-1]
            idx+=1
        # print(pre)
        idx = n-2
        while idx>=0:
            post[idx] = post[idx+1]+nums[idx+1]
            idx-=1
        # print(post)
        idx = 0
        while idx<n:
            ans[idx] = abs(pre[idx]-post[idx])
            idx+=1
        return ans